nipy/algorithms/statistics/formula/formulae.py

```diff
@@ -575,33 +585,47 @@ def __add__(self, other):
         """
         if not is_formula(other):
             raise ValueError('only Formula objects can be added to a Formula')
-        f = self.__class__(np.hstack([self.terms, other.terms]))
+        f = Formula(np.hstack([self.terms, other.terms]))
```

Comment:
The problem was that the add and sub methods create a new object at the end, from a list of terms.  By using ``self.__class__`` to construct the output object, I am asserting that the class constructor has the same signature as the Formula class, and that is not true of Factor, which needs a name.

I guess one way round that would be to have a ``from_terms`` method to Formula, and use that, but in fact I don't think its unreasonable to return a Formula from the Factor subtraction, the main logic of the Factor is in the constructor, and combining two factors can make something that isn't a pure Factor